backend/src/db/migrations/20230826083159_add_user_roles_table.ts

```diff
@@ -0,0 +1,53 @@
+import { type Knex } from 'knex';
+
+const TABLE_NAME = 'user_roles';
+
+const COLUMN_NAME = {
+  ID: 'id',
+  NAME: 'name',
+  KEY: 'key',
+  CREATED_AT: 'created_at',
+  UPDATED_AT: 'updated_at',
+} as const;
+
+const USER_NAMES = {
+  CHATBOT: 'Chatbot',
+  USER: 'User',
+} as const;
+
+const USER_ROLES = {
+  CHATBOT: 'chatbot',
+  USER: 'user',
+} as const;
+
+const ROLES = [
+  {
+    name: 'Chatbot',
+    key: 'chatbot',
+  },
+  { name: 'User', key: 'user' },
+];
+
+async function up(knex: Knex): Promise<void> {
+  await knex.schema.createTable(TABLE_NAME, (table) => {
+    table.increments(COLUMN_NAME.ID).primary();
+    table.enum(COLUMN_NAME.NAME, Object.values(USER_NAMES));
+    table.enum(COLUMN_NAME.KEY, Object.values(USER_ROLES));
```

Comment:
must be strings and not enums, there is no point in enum because we don't have predefined values and all values will be unique. This works, for example, if we want to set statuses and we know that we will only have three statuses, which can be repeated in some rows. But in this case, if we want to add a new role, it will never satisfy this enum and we will need to always expand it

Reply:
yes, ok, I returned it back